cityscapesscripts/evaluation/evalObjectDetection3D.py

```diff
@@ -0,0 +1,1186 @@
+#!/usr/bin/python3
+#
+# The evaluation script for Cityscapes 3D object detection (https://arxiv.org/abs/2006.07864)
+# We use this script to evaluate your approach on the test set.
+# You can use the script to evaluate on the validation set.
+#
+
+# python imports
+import coloredlogs
+import logging
+import numpy as np
+import json
+import os
+import argparse
+from typing import (
+    List,
+    Tuple
+)
+
+from pyquaternion import Quaternion
+from tqdm import tqdm
+from copy import deepcopy
+import concurrent.futures
+
+from cityscapesscripts.helpers.labels import labels
+from cityscapesscripts.evaluation.objectDetectionHelpers import (
+    Box3DObject,
+    IgnoreObject,
+    EvaluationParameters,
+    getFiles,
+    calcIouMatrix,
+    calcOverlapMatrix
+)
+from cityscapesscripts.evaluation.objectDetectionHelpers import (
+    MATCHING_MODAL,
+    MATCHING_AMODAL
+)
+from cityscapesscripts.evaluation.plot3DResults import (
+    prepare_data,
+    plot_data
+)
+
+logger = logging.getLogger('EvalObjectDetection3D')
+logging.basicConfig(filename='eval.log',
+                    filemode='w',
+                    format='%(asctime)s.%(msecs)d %(name)s %(levelname)s %(message)s',
+                    datefmt='%H:%M:%S')
+coloredlogs.install(level='INFO')
+
+"""
+The evaluation script expects one json annotation
+file per image with the format:
+
+{
+    "annotation": [
+        {
+            "2d": {
+                "modal": [xmin, ymin, xmax, ymax],
+                "amodal": [xmin, ymin, xmax, ymax]
+            },
+            "3d": {
+                "center": [x, y, z],
+                "dimensions": [length, width, height],
+                "rotation": [q1, q2, q3, q4],
+                "format": "CRS_ISO8855"
+            },
+            "class_name": str,
+            "score": 1.0
+        }
+    ]
+}
+
+"""
+
+def printErrorAndExit(msg):
+    logger.error(msg)
+    logger.info("========================")
+    logger.info("=== Stop evaluation ====")
+    logger.info("========================")
+
+    exit(1)
+
+class Box3DEvaluator:
+    """The Box3DEvaluator object contains the data as well as the parameters
+    for the evluation of the dataset.
+
+    :param eval_params: evaluation params including max depth, min iou etc.
+    :type eval_params: EvaluationParameters
+    :param gts: all GT annotations per image
+    :type gts: dict
+    :param preds: all GT annotations per image
+    :type preds: dict
+    :param ap: data for Average Precision (AP) calculation
+    :type ap: dict
+    :param results: evaluation results
+    :type results: dict
+    """
+
+    def __init__(
+        self,
+        evaluation_params: EvaluationParameters
+        ) -> None:
+
+        self.eval_params = evaluation_params
+        self._num_steps = 50
```

Comment:
Why is this not in EvaluationParameters?

Reply:
moved to EvaluationParameters.